services/ark-api/ark-api/src/ark_api/auth/middleware.py

```
@@ -33,79 +39,131 @@
-        # and OIDC configuration is available
-        auth_mode = os.getenv("AUTH_MODE", "").lower()
+        # Get authentication mode and OIDC configuration
+        auth_mode_raw = os.getenv("AUTH_MODE", "").lower()
         oidc_issuer = os.getenv("OIDC_ISSUER_URL", "")
         oidc_app_id = os.getenv("OIDC_APPLICATION_ID", "")
         
-        # Check OIDC configuration and log appropriate messages
-        if auth_mode == "sso":
-            if not oidc_issuer:
-                logger.warning("AUTH_MODE=sso but  is not configured. Authentication disabled.")
-            elif not oidc_app_id:
-                logger.warning("AUTH_MODE=sso but OIDC_APPLICATION_ID is not configured. Authentication disabled.")
-            else:
-                logger.debug("Authentication enabled: AUTH_MODE=sso with valid OIDC configuration")
+        # Validate auth mode - default to "open" for invalid values
+        valid_auth_modes = [AuthMode.SSO, AuthMode.BASIC, AuthMode.HYBRID, AuthMode.OPEN]
+        if auth_mode_raw in valid_auth_modes:
+            auth_mode = auth_mode_raw
         else:
-            logger.debug(f"Authentication disabled: AUTH_MODE={auth_mode or 'not set'}")
+            auth_mode = AuthMode.OPEN
+            if auth_mode_raw:  # Only log if a value was actually set
+                logger.warning(f"Invalid AUTH_MODE '{auth_mode_raw}', defaulting to 'open'. Valid values: {valid_auth_modes}")
         
-        # Skip authentication if AUTH_MODE is not "sso" or OIDC config is missing
-        skip_auth = (auth_mode != "sso") or (not oidc_issuer) or (not oidc_app_id)
-
-        if skip_auth:
+        # Log authentication configuration
+        logger.debug(f"Auth mode: {auth_mode} (raw: {auth_mode_raw}), Path: {path}")
+        
+        # Determine which auth methods are enabled
+        jwt_enabled = auth_mode in [AuthMode.SSO, AuthMode.HYBRID] and oidc_issuer and oidc_app_id
```

I dont think `and` oidc_issuer and oidc_app_id` is needed. If someone sets the auth mode to one of these but does not provide an oidc issuer or app_id then thats a configuration issue and the jwt should still be enabled, no? I might not see this through properly

---

Not sure I understand. If OIDC configuration is missing when auth mode is set to SSO/HYBRID, the application should fail fast and clearly indicate the configuration error. This prevents runtime issues and forces proper configuration upfront

---

I agree, I just don't see how this makes it fail